### PR TITLE
[SAM] Restore Gauge  + Fuko Bug Fix

### PR DESCRIPTION
**Pull request by Rjb99**
This restores all of SAM's Sen and Kenki Gauge to the Analysis.
Also includes a "Bugfix" to Fuga/Fuko by erasing Fuga from the pages of history

### CODE REVIEW

File: src/parser/jobs/sam/modules/Sen.tsx [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+		}
+	}
+
+	// Function that handles SenState check, if no senState call the maker and then push to the rotation
+	checkCastAndPush(event: Events['action']) : void {
+		// check the sen state, if undefined/not active, make one
+
+		let lastSenState = this.lastSenState
+
+		if ((typeof lastSenState === 'undefined') || (lastSenState.isDone === true)) {
```
Comment (AzariahAeroborn) [inline src/parser/jobs/sam/modules/Sen.tsx]:
```suggestion
		if ((lastSenState == null) || (lastSenState.isDone === true)) {
```
Please use == null (or != as necessary) when checking for null/undefined values.

File: src/parser/jobs/sam/modules/Kenki/Kenki.tsx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+	private eyeExam(event: Events['damage']) {
+		const targetedSelf = event.targets.some(({target}) => target === this.parser.actor.id)
+		if (targetedSelf) {
+			this.kenkiGauge.modify(THIRD_EYE_GAIN)
+		}
+	}
+
+	private onRemoveEye() {
+		if (this.damageHook !== undefined) {
```
Comment (AzariahAeroborn) [inline src/parser/jobs/sam/modules/Kenki/Kenki.tsx]:
```suggestion
		if (this.damageHook != null) {
```

File: src/parser/jobs/sam/modules/Kenki/Kenki.tsx [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+			this.removeEventHook(this.damageHook)
+		}
+	}
+
+	private onGaugeModifier(event: Events['action' | 'combo']) {
+		const modifier = this.kenkiGaugeModifiers.get(event.action)
+
+		if (modifier !== undefined)	{
```
Comment (AzariahAeroborn) [inline src/parser/jobs/sam/modules/Kenki/Kenki.tsx]:
```suggestion
		if (modifier != null)	{
```

File: src/parser/jobs/sam/modules/index.js [hunk 4]
```diff
@@ -2,20 +2,24 @@ import {AoeChecker} from './AoeChecker'
 import {Buffs} from './Buffs'
 import {Combos} from './Combos'
 import {Fuka} from './Fuka'
-// import {Higanbana} from './Higanbana'
 import Interrupts from './Interrupts'
+import {Kenki} from './Kenki'
+// import {Higanbana} from './Higanbana'
```
Comment (AzariahAeroborn) [inline src/parser/jobs/sam/modules/index.js:7]: commented out so it doesn't really matter here, but this should still be above Interrupts to maintain alphabetical order of imports

**Review by AzariahAeroborn — CHANGES_REQUESTED** [review]

**Comment by Rjb99** [comment]
Requested Changes made. Import order will be fixed in the next PR with Higanabana

**Review by AzariahAeroborn — APPROVED** [review]

**Pull request merged**